emcc.py

```diff
@@ -1439,6 +1442,12 @@ def check(input_file):
             passes += ['--strip-debug']
           if not shared.Settings.EMIT_PRODUCERS_SECTION:
             passes += ['--strip-producers']
+          if shared.Settings.AUTODEBUG and shared.Settings.WASM_OBJECT_FILES:
+            # adding '--flatten' here may make these even more effective
+            passes += ['--instrument-locals']
+            passes += ['--log-execution']
+            passes += ['--instrument-memory']
+            passes += ['--legalize-js-interface']
```

Comment:
I guess most of the heavy lifting to modify the code is done here in binaryen?

Reply:
Yeah, these binaryen passes do all the actual instrumentation. Basically this PR automates the JS side of adding the necessary stuff for that instrumentation to be useful.